simple WGLC review suggestions (#237)
ietf-rats-wg/architecture: draft-ietf-rats-architecture.md

## Network Endpoint Assessment

Network operators want a trustworthy report that includes identity

> and version information about the hardware and software on the

Author: I am not sure I understand what "version _of_ information" means.  I thought it was a typo.

> and version information about the hardware and software on the

Reviewer: group agrees that the new text was what was intended.

provides device identity and protected storage for measurements.
The system components perform a series of measurements that may be
signed by the root of trust, considered as Evidence about the hardware,

> firmware, BIOS, software, etc. that is present.

Reviewer 2: I agree with this change, since one doesn't think of hardware as "running" per se.

This is a generalization of the ML model use case above, where
the data can be any highly confidential data, such as health data
about customers, payroll data about employees, future business plans, etc.
As part of the attestation procedure, an assessment is made against a set

> of policies to evaluate the state of the system which is requesting

Reviewer 2: ```suggestion
of policies to evaluate the state of the system that is requesting
```
("that" and "which" are generally interchangeable in British English, but in American there is a subtle difference per Chicago Manual of Style.  The RFC Edit pass likes to correct these so pre-emptively doing so here.)

might be combined.
Other implementations might have multiple Attesting and Target Environments,
such as in the examples described in more detail in {{layered-attestation}}
and {{compositedevice}}.  Other examples may exist.  Besides, the examples
discussed could be combined into even more complex implementations.

Claims are collected from Target Environments, as shown in {{twotypes-env}}.

> That is, Attesting Environments collect the values and the information to be represented in Claims, by reading system registers and variables, calling into subsystems, taking measurements on code, memory or other security related assets of the Target Environment.

Reviewer: ```suggestion
That is, Attesting Environments collect the values and the information to be represented in Claims, by reading system registers and variables, calling into subsystems, taking measurements on code, memory, or other security related assets of the Target Environment.
```

has to ensure the integrity of the bootloader (Target Environment B).
There are
potentially multiple kernels to boot, and the decision is up to the bootloader.

> Only a bootloader with intact integrity will make an appropriate decision. Therefore, the relevant Claims have to be measured securely.

Reviewer: ```suggestion
Only a bootloader with intact integrity will make an appropriate decision. 
Therefore, the Claims relating to the integrity of the bootloader have to be measured securely.
```

## Attester

In some scenarios, Evidence might contain sensitive information such as

> Personally Identifiable Information (PII).

Reviewer: ```suggestion
Personally Identifiable Information (PII) or system identifiable information.
```

Thus, an Attester must trust entities to which it conveys Evidence, to not
reveal sensitive data to unauthorized parties.

> The Verifier might share this information with other authorized parties, according to predefined rules that address the handling of PII (potentially included in Appraisal Policies for Evidence).

Reviewer: ```suggestion
The Verifier might share this information with other authorized parties, according to a governing policy that address the handling of sensitive information (potentially included in Appraisal Policies for Evidence).
```

> Personally Identifiable Information.

Reviewer: ```suggestion
Personally Identifiable Information (PII).
```

Thus, an Attester must trust entities to which it conveys Evidence, to not
reveal sensitive data to unauthorized parties.

> The Verifier might share this information with other authorized parties, according to some predefined rules.

Author: it's not the colloquial flavour that I wanted to fix here, it's the fact that the Verifier _controls_ the rules for PII sharing that seemed wrong.

> The Verifier might share this information with other authorized parties, according to some predefined rules.

Reviewer 3: Ack. I'd make it more specific then:

```suggestion
The Verifier might share this information with other authorized parties, according to predefined rules that address the handling of PII (potentially included in Appraisal Policies for Evidence).
```

In the background-check model, this Evidence may also be revealed to Relying Party(s).

> When Evidence contains sensitive information, an Attester

Reviewer 3: Seems the intend here was to say "this is not always the case when sensitive information is included, but sometimes".

Also, what is sensitive information in the first place? Does sensitive information require extensive integrity protection, because it breaks easily? Seems a tad bit too colloquial to me - not only here, but in the surrounding text.

Author: > Seems the intend here was to say "this is not always the case when sensitive information is included, but sometimes".

I added "typically" below because while this is optional, it makes sense to take some precaution in this specific case.

> Also, what is sensitive information in the first place? Does sensitive information require extensive integrity protection, because it breaks easily? Seems a tad bit too colloquial to me - not only here, but in the surrounding text.

This is in the context of PII handling between mutually trusting parties, which is established by the preceding paragraph.

typically requires that a Verifier authenticates itself (e.g., at TLS session establishment) and might even request a remote attestation before the Attester

> sends the sensitive Evidence.  This can be done by having the

Reviewer 3: Text is improved, but I am not really happy with the text in general, as of yet.

and a manufacturer-signed IDevID certificate) secured in the Attester.

The Verifier is provided with an appropriate trust anchor, or provided with a database of public keys (rather than certificates), or even carefully secured lists of symmetric keys.

> The nature of how the Verifier manages to validate the signatures produced by the Attester is critical to the secure operation of an Attestation system, but is not the subject of standardization within this architecture.

Reviewer 3: +1 grammar

Attestation Results are the input used by the Relying Party to decide the extent to which it will trust a particular Attester, and allow it to access some data or perform some operation.

> Attestation Results may carry a boolean value indicating compliance or non-compliance with a Verifier's appraisal policy, or a richer set of Claims about the Attester, against which the Relying Party applies its Appraisal Policy for Attestation Results.

Reviewer 3: ```suggestion
Attestation Results may be expressed via boolean values indicating compliance or non-compliance with a Verifier's appraisal policy, or via a richer set of Claims about the Attester, against which the Relying Party applies its Appraisal Policy for Attestation Results.
```

### On-Device Attester and Key Protection

> It is assumed that the Attester is located in an isolated environment of a device like a process, a dedicated chip, a TEE or such that collects the Claims, formats them and signs them with an Attestation Key. The Attester must be protected from unauthorized modification to ensure it behaves correctly. There must also be confidentiality so that the signing key is not captured and used elsewhere to forge evidence.

Reviewer: ```suggestion
It is assumed that the Attester is located in an isolated environment of a device like a process, a dedicated chip a TEE or such that collects the Claims, formats them and signs them with an Attestation Key. The Attester must be protected from unauthorized modification to ensure it behaves correctly. There must also be confidentiality so that the signing key is not captured and used elsewhere to forge evidence.
```

the integrity of the entity providing the appraisal policy needs to be
considered.

> The security of conveyed information may be applied at different layers, whether by a conveyance protocol, or an information encoding format. This architecture expects attestation messages (i.e., Evidence, Attestation Results, Endorsements, Reference Values and Policies) are end-to-end protected based on the role interaction context.

Reviewer 3: +1 nice catch

> The security of conveyed information may be applied at different layers, whether by a conveyance protocol, or an information encoding format. This architecture expects attestation messages (i.e., Evidence, Attestation Results, Endorsements, Reference Values and Policies) are end-to-end protected based on the role interaction context.

Reviewer 2: ```suggestion
The security of conveyed information may be applied at different layers, whether by a conveyance protocol, or an information encoding format. This architecture expects attestation messages (i.e., Evidence, Attestation Results, Endorsements, Reference Values, and Policies) are end-to-end protected based on the role interaction context.
```
